tests/contrib/codemeta/conftest.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""CodeMeta specific test configs."""
+import pytest
+from invenio_vocabularies.proxies import current_service as vocabulary_service
+from invenio_vocabularies.records.api import Vocabulary
+
+from invenio_rdm_records.cli import create_records_custom_field
+from invenio_rdm_records.contrib.codemeta.custom_fields import (
+    CODE_CUSTOM_FIELDS,
+    CODE_FACETS,
+    CODE_NAMESPACE,
+)
+from invenio_rdm_records.proxies import current_rdm_records_service
+
+
+@pytest.fixture(scope="module")
+def service():
+    """Rdm records service instance."""
+    return current_rdm_records_service
+
+
+@pytest.fixture()
+def codemeta_development_status(superuser_identity):
+    """Creates and retrieves a vocabulary type."""
+    v = vocabulary_service.create_type(superuser_identity, "developmentstatus", "ds")
+    return v
+
+
+@pytest.fixture()
+def codemeta_development_status_vocabulary(
+    superuser_identity, codemeta_development_status
+):
+    """Creates and retrieves controlled vocabulary 'development status'."""
+    record = vocabulary_service.create(
+        identity=superuser_identity,
+        data={
+            "id": "concept",
+            "title": {
+                "en": "Concept",
+            },
+            "description": {
+                "en": "Minimal or no implementation has been done yet, or the repository is only intended to be a limited example, demo, or proof-of-concept."
+            },
+            "type": "developmentstatus",
+        },
+    )
+    Vocabulary.index.refresh()  # Refresh the index
+
+    return record
+
+
+@pytest.fixture(scope="function")
+def init_codemeta_custom_fields(
+    running_app,
+    cli_runner,
+    codemeta_development_status_vocabulary,
+    codemeta_cf_facetted_term,
+):
+    """Initializes codemeta custom fields."""
+    # Add code meta custom fields
+    running_app.app.config["RDM_CUSTOM_FIELDS"] = CODE_CUSTOM_FIELDS
+    running_app.app.config["RDM_NAMESPACES"] = CODE_NAMESPACE
+    loaded_facets = running_app.app.config["RDM_FACETS"]
+    running_app.app.config["RDM_FACETS"] = {**loaded_facets, **CODE_FACETS}
+    loaded_search = running_app.app.config["RDM_SEARCH"]
+    running_app.app.config["RDM_SEARCH"] = {
+        **loaded_search,
+        "facets": loaded_search["facets"] + [codemeta_cf_facetted_term],
+    }
+    result = cli_runner(create_records_custom_field)
+    assert result.exit_code == 0
+
+
+@pytest.fixture(scope="function")
```

Comment:
this fixtures will be recreated per test function there are mere constants, using `session` would improve performance. You can also just declare a global variable.